rand_core/src/lib.rs

```diff
@@ -400,90 +478,82 @@ pub trait SeedableRng: Sized {
 impl<'a, R: RngCore + ?Sized> RngCore for &'a mut R {
     #[inline(always)]
     fn next_u32(&mut self) -> u32 {
-        (**self).next_u32()
+        R::next_u32(self)
     }
 
     #[inline(always)]
     fn next_u64(&mut self) -> u64 {
-        (**self).next_u64()
-    }
-
-    #[inline(always)]
-    fn fill_bytes(&mut self, dest: &mut [u8]) {
-        (**self).fill_bytes(dest)
+        R::next_u64(self)
     }
 
     #[inline(always)]
-    fn try_fill_bytes(&mut self, dest: &mut [u8]) -> Result<(), Error> {
-        (**self).try_fill_bytes(dest)
+    fn fill_bytes(&mut self, dst: &mut [u8]) {
+        R::fill_bytes(self, dst)
     }
 }
 
+// Implement `TryCryptoRng` for references to a `TryCryptoRng`.
+impl<'a, R: CryptoRng + ?Sized> CryptoRng for &'a mut R {}
+
 // Implement `RngCore` for boxed references to an `RngCore`.
 // Force inlining all functions, so that it is up to the `RngCore`
 // implementation and the optimizer to decide on inlining.
 #[cfg(feature = "alloc")]
 impl<R: RngCore + ?Sized> RngCore for Box<R> {
     #[inline(always)]
     fn next_u32(&mut self) -> u32 {
-        (**self).next_u32()
+        R::next_u32(self)
     }
 
     #[inline(always)]
     fn next_u64(&mut self) -> u64 {
-        (**self).next_u64()
+        R::next_u64(self)
     }
 
     #[inline(always)]
     fn fill_bytes(&mut self, dest: &mut [u8]) {
-        (**self).fill_bytes(dest)
-    }
-
-    #[inline(always)]
-    fn try_fill_bytes(&mut self, dest: &mut [u8]) -> Result<(), Error> {
-        (**self).try_fill_bytes(dest)
+        R::fill_bytes(self, dest)
     }
 }
 
+// Implement `TryCryptoRng` for boxed references to a `TryCryptoRng`.
```

Comment:
This comment isn't useful IMO. Same above.

Shame we can't add the actual `TryCryptoRng` impls.

Reply:
Removed.